src/tool_application_support.cpp

```diff
@@ -27,38 +27,64 @@ namespace clangmetatool {
 
 namespace {
 
-std::string getExecutablePath(const std::string &argv0) {
+std::string getAbsoluteRealpath(const std::string &path) {
+  std::string absRealpath = path;
+  if (llvm::sys::path::is_relative) {
+    llvm::SmallString<128> absRealpathVec(path);
+    auto err = llvm::sys::fs::make_absolute(absRealpathVec);
+    if (err) {
+      return {};
+    }
+    absRealpath = absRealpathVec.str().str();
+  }
+  llvm::SmallString<128> realpath;
+  auto err = llvm::sys::fs::real_path(absRealpath, realpath, true);
+  if (err) {
+    return {};
+  }
+  return realpath.str().str();
+}
+
+std::string getExecutablePathFromArgv(const std::string &argv0) {
   // Use the address of 'main' to locate the executable name, it is possible
   // that this may return an empty address
   std::string exePath =
       llvm::sys::fs::getMainExecutable(argv0.c_str(), nullptr);
-  if (!exePath.empty()) {
-    clang::DiagnosticsEngine diagnostics(new clang::DiagnosticIDs,
-                                         new clang::DiagnosticOptions,
-                                         new clang::IgnoringDiagConsumer);
-    clang::driver::Driver driver(exePath, llvm::sys::getDefaultTargetTriple(),
-                                 diagnostics);
-    if (llvm::sys::fs::exists(driver.ResourceDir)) {
-      return exePath;
-    }
+  exePath = getAbsoluteRealpath(exePath);
+  if (exePath.empty()) {
+    // Attempt to use argv0
+    exePath = getAbsoluteRealpath(argv0);
```

Comment:
You don't need to call realpath multiple times and I think you should check upfront here if this has a path or not. Like:

```
exePath = argv0;
if (!llvm::sys::path::has_parent_path(exePath)) {
    llvm::Optional<std::string> maybeExePath =
    llvm::sys::Process::FindInEnvPath("PATH", argv0);
    exePath = maybeExePath.getValueOr("");
}

return !exePath.empty() ? getRealpath(exePath) : "";
```